### ISSUE TITLE
App Requires User Action Despite Opponent's Time Expiring

### ISSUE THREAD

**Issue by opsrcode** [lichess-org/mobile#730]
### What is the problem?

During a game against any opponent, when the opponent leaves the match before the time runs out, the app will present the options "claim victory" or "claim draw" as usual. However, if this happens moments before the end of the game, when the opponent's time is about to run out, the message remains even after the opponent's time expires. The options remain functional, allowing the user (me) to choose an option to alter the result.

### Steps to reproduce the bug

1. Open the Lichess app.
2. Start a game against any opponent.
3. Play the game until the opponent's time is nearly expired.
4. Observe if the opponent leaves the match just before their time runs out.
5. Notice that the app presents the options "claim victory" or "claim draw."
6. Observe that the message with the options "claim victory" or "claim draw" remains visible and requires user interaction to finalize the game result, even though the opponent's time has already expired.

### What did you expect to happen?

The app should ensure a victory or result automatically since the opponent's time has expired, as it happens on the browser platform.

### What happened instead?

The app waits for the user's choice even after the opponent's time has run out.

### Device Info

Galaxy A21s

### Operating System Version

Android version 12

### Additional information

Knox version 3.8
Knox API level 35
Lichess Beta version 0.8.10 (Play Store, updated on May 16, 2024)
![Screenshot_20240531-020309](https://github.com/lichess-org/mobile/assets/141454158/4fd86f7e-8005-4a8a-91b9-51d425ee1a41)

**Comment by opsrcode** [comment]
Once again, the same bug occurred in the version downloaded via Github v0.8.17:
![Screenshot_20240608-144709.jpg](https://github.com/lichess-org/mobile/assets/141454158/1eace237-4a1d-4eb3-94fa-69487c71c0be)

**Comment by opsrcode** [comment]
I conducted some tests on the bug I found earlier, which can be seen in the messages above, and I noticed new features that were affected by the same bug. Tests were conducted, and the screenshots below from the Android app show these different actions in the app's functionality.

![Screenshot_20240612-152134.jpg](https://github.com/lichess-org/mobile/assets/141454158/2d3aec82-5e37-4199-ba7c-134f101f649f)

The screenshot above represents the screen when the bug occurred: the user left the match, and the time ran out, but the app did not give me a response.

![Screenshot_20240612-152216.jpg](https://github.com/lichess-org/mobile/assets/141454158/457ebd09-d585-4707-b945-ccc5c93ea739)

In the screenshot above, you can see what happens when the user clicks outside the options suggested by the app. By clicking on the common match options generated during the game, we can see that the app still prompts the user to claim victory or accept a draw, indicating that the match has not yet ended.

![Screenshot_20240612-152349.jpg](https://github.com/lichess-org/mobile/assets/141454158/f2ef08eb-c50c-4c52-bef8-9db65b3f4b02)

Another test conducted was closing the application during the bug, while the app's request was active. As shown in the screenshot above, the match where the user left and the time ran out is not recorded in the recent results.

![Screenshot_20240612-152452.jpg](https://github.com/lichess-org/mobile/assets/141454158/49bd90c1-4516-4f89-bb3d-6ee9d1770921)

Now we move to the browser platform, accessed via Android. Through the browser, we can see that the match appears as ongoing.

![Screenshot_20240612-152512.jpg](https://github.com/lichess-org/mobile/assets/141454158/064265e4-eaeb-469e-a24b-074dac74f9a3)

And finally, when we enter the match that is theoretically ongoing, we see that on the browser platform, it gives us the correct result of the match, which is a win due to the opponent's time running out.

**Comment by ghost** [comment]
@veloce This was solved. I don't know when or how, but today I encountered the situation again, and it now works as intended in the latest version (v0.17.8).

![Image](https://github.com/user-attachments/assets/517e9b74-aa4f-481b-bece-ca3dcdc1d829)

**Comment by ghost** [comment]
@veloce But when clicking outside of this rectangle, the message to claim a draw or a win will still be present, just non-interactive. I couldn't take a screenshot because I accidentally tapped another corner of the screen haha. But I believe it should be reproducible again.

**Comment by MaartenD** [comment]
@veloce i reproduced this issue today (see attachment) and in my opinion this issue is solved. If you agree, would you be so kind to close this issue?

https://github.com/user-attachments/assets/eccf4e65-afe3-4ab3-b3a6-8f12a44dd638

**Comment by veloce** [comment]
So if you think this issue is closed it mean you did *not* reproduce the issue, right @MaartenD ?

**Comment by MaartenD** [comment]
@veloce correct. The error can't be reproduced (see video attachment in my previous post). To be clear, in my opinion the app is working correct.
